### PR TITLE
Add support subordinate user/group ids as separate entries

### PR DESCRIPTION
**Pull request by tiran**
New LDAP object class "ipaSubordinateIdEntry" + additional classes with five new attributes:
- ipasubuidnumber / ipasubuidcount
- ipasubgidnumber / ipasgbuidcount
- ipaOwner

New plugin ``subid`` to manage subordinate ids with public commands subid-generate, subid-match, subid-find, subid-show. The commands subid-add and subid-del are internal commands. Users lack permissions to add/del entries.

New permissions and self-service permissions

The code hard-codes counts to 65536, sets subgid equal to subuid, and
does not allow removal of subids. There is also a hack that emulates a
DNA plugin with step interval 65536 for testing.

## TODO

- [X] bike shed on command names ``subid-generate`` and ``subid-match``. Are these names self-explaining and easy to remember?
- [ ] Decide whether we should allow removing subids from users. I decided against it. Subid space is limited and we cannot safely recycle a subid any way.
- [ ] Include a helper script to import subuid and subgids from an existing databases? Customers may have existing ``/etc/subuid`` and ``/etc/subgid`` files.
- [ ] Enforce / check for subid interval overlaps? https://github.com/freeipa/freeipa/pull/5438 has some hacky code.
- [X] Add helper to show how many subid intervals are left, it's ``subid-stats``
- [ ] Should we add a unique check on ``ipaSubUidNumber`` and ``ipaSubGidNumber``?
- [x] explain that IPA's random baseid does not conflict with subid range start.
- [x] Add example how subids are allocated and assign.
- [x] Investigate new ID range type. First attempt to add ``ipa-local-subid`` failed.
- [x] sidgen task fails ``Failed to convert LDAP entry to range struct`` in ``ipa_sidgen_common.c`` because subid_range has no RIDs. Added hack to include RIDs by default.
- [x] Failing sudo test case ``test_domain_resolution_order`` is tracked at https://github.com/SSSD/sssd/issues/5571
- [ ] submit OIDs to RHANANA
- [ ] Mention that systemd's dynamic user feature uses range uids in range 61184 to 65519.
- [ ] Decide how to deal with ``user-del --preserve``. (see design doc for more details)
- [x] Improve WebUI
   - [x] Replace subid-add with subid-generate in user memberOf view
   - [x] Replace subid-add with subid-generate in subid view
   - [x] Show subid-stats
   - [x] Hide delete
   - [x] i18n

See: https://pagure.io/freeipa/issue/8361
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

**Comment by flo-renaud** [comment]
@tiran FYI the tox issue is a known issue: [8818](https://pagure.io/freeipa/issue/8818) and there is a PR in progress: [5737](https://github.com/freeipa/freeipa/pull/5737)

**Comment by tiran** [comment]
PR will fail to build until updated 389-DS builds are available

File: install/share/60ipaconfig.ldif [hunk 1]
```diff
@@ -45,11 +46,13 @@ attributeTypes: ( 2.16.840.1.113730.3.8.3.26 NAME 'ipaSELinuxUserMapDefault' DES
 attributeTypes: ( 2.16.840.1.113730.3.8.3.27 NAME 'ipaSELinuxUserMapOrder' DESC 'Available SELinux user context ordering' EQUALITY caseIgnoreMatch ORDERING caseIgnoreMatch SUBSTR caseIgnoreSubstringsMatch SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE X-ORIGIN 'IPA v3')
 ## ipaMaxHostnameLength - maximum hostname length to allow
 attributeTypes: ( 2.16.840.1.113730.3.8.1.28 NAME 'ipaMaxHostnameLength' EQUALITY integerMatch SYNTAX 1.3.6.1.4.1.1466.115.121.1.27 SINGLE-VALUE)
+# ipaUserDefaultSubordinateId - if TRUE new user entries gain subordinate id by default
+attributeTypes: ( 2.16.840.1.113730.3.8.3.23.14 NAME 'ipaUserDefaultSubordinateId' DESC 'Enable adding user entries with subordinate id' SYNTAX 1.3.6.1.4.1.1466.115.121.1.7 SINGLE-VALUE X-ORIGIN 'IPA v4.9')
```
Comment (flo-renaud) [inline install/share/60ipaconfig.ldif:50]: The OID contains an extra .3: .8.3.23.14 should be .8.23.14

Comment (tiran) [reply install/share/60ipaconfig.ldif]: Thanks, the comment was correct but the definition was wrong.

**Review by fcami — APPROVED** [review]
ACK for when Azure CI passes

**Comment by rcritten** [comment]
NOTE to merger: remove the ack from the ipa-4-9 backport so it isn't merged until we can confirm when the next upstream release is.

**Comment by flo-renaud** [comment]
Azure is failing in pycodestyle:
pycodestyle
-----------
./ipaserver/install/ldapupdate.py:360:1: W293 blank line contains whitespace

**Comment by rcritten** [comment]
Maybe I'm going insane but I pulled this branch and I don't see any trailing whitespace on line 360.

**Comment by flo-renaud** [comment]
@rcritten : @tiran already fixed the trailing whitespace, and PRCI is now green :)

**Comment by rcritten** [comment]
master:

* 1c4ae37293c31c5030556e8464099cad22d49db8 Add basic support for subordinate user/group ids
* c78d1341ad54a1837f64ee6a332f534ecd534046 Redesign subid feature
* 51035d9354e0f7497bbdd5486b4717b385ee5a62 Use 389-DS' dnaInterval setting to assign intervals
* 1e00748f1833a79488aeaa9ba798ba6538905fa7 Fix ipa-server-upgrade
* 110940b998b0d7adb68fa097dd973b67bbdbd644 Fix oid of ipaUserDefaultSubordinateId
* 30eceb504adcfd7e223430753fb45bf47fb9673b WebUI: Improve subordinate ids user workflow
* aae6c029dac285c1704fea3d97e576f81d0bc357 Test DNA plugin configuration

**Comment by rcritten** [comment]
@tiran a manual backport is needed for ipa-4-9, can you create that?

**Pull request closed without merging**